atomix/cluster/src/main/java/io/atomix/raft/roles/PassiveRole.java

```diff
@@ -391,6 +410,10 @@ private void abortPendingSnapshots() {
 
     // Append the entries to the log.
     appendEntries(request, future);
+
+    // If a snapshot replication was ongoing, reset it. Otherwise SnapshotReplicationListeners will
+    // wait for ever for the snapshot to be received.
+    resetOngoingSnapshotReplication.run();
```

Comment:
:thought_balloon: If only we had a state machine :smile:

Reply:
:question: Actually, more generally, shouldn't we abort any pending snapshot if we receive an append request in between? I'm surprised we were not already doing this. We should consider it.

Reply:
I don't get what you mean. We were not aborting pending snapshot when receiving an append request. But now we are doing it. What should we consider?

Reply:
Sorry, I meant exactly what you said. So maybe we should consider calling `abortPendingSnapshots` here instead of the reset?

Reply:
I have now removed the reset and call `abortPendingSnapshots` here.